src/Generator/TypeGenerator.php

```diff
@@ -73,6 +78,32 @@ public static function fromTypeString($type)
         return $instance;
     }
 
+    /**
+     *  @param array $typeArray
+     *
+     *  @return TypeGenerator
+     *
+     *  @throws InvalidArgumentException
+     */
+    public static function fromTypeArray(array $typeArray)
+    {
+        if (!isset($typeArray['name'])) {
+            throw new InvalidArgumentException(sprintf(
+                'Provided type "%s" is invalid: must conform "%s"',
+                var_export($typeArray, true),
+                self::$validIdentifierMatcher
+            ));
+        }
+
+        $instance = self::fromTypeString($typeArray['name']);
+
+        if (isset($typeArray['alias']) && is_bool($typeArray['alias'])) {
```

Comment:
Do we want to throw an `InvalidArgumentException` if the alias value is not of boolean type?